### PR TITLE
Feature/71466 ui new library module in preservation framework

### PR DESCRIPTION
**Pull request by arildbjerkenes**
At this point we should not try to implement the complete library framework for library. There are a lot of things to consider and discuss. For now we implement enough to be able to support what we need for Preservation. 

The treeview component is not complete at this point. We should include that a node is shown as 'selected', and the component should expose the path to the selected node (e.g. array with node-names). Maybe we should store an object representing the node (the object we fetch from server). 

There are some 'prof of concept' code here as well. 

Currently the node selected is not reflected in the url. My plan was that instead of changing a state variable, we should do a routing, when a node is selected. I suggest we add a new PBI to add this.

### CODE REVIEW

File: src/modules/PlantConfig/style.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    font-family: Equinor;
+    font-style: normal;
+    color: var(--text--primary-white);
+    font-size: 14px;
+    padding: 10px 16px;
+    border-radius: 4px;
+    border: none;
+    background-color: ${tokens.colors.interactive.primary__resting.rgba};
+    font-weight: 500;
+`;
```
Comment (kfaln) [inline src/modules/PlantConfig/style.ts]: This is probably not used, here and also in `Preservation/style.ts`

File: src/modules/PlantConfig/views/Library/Library.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from 'react';
+//import withAccessControl from '../../../../core/security/withAccessControl';
```
Comment (kfaln) [inline src/modules/PlantConfig/views/Library/Library.tsx]: Remove

File: src/modules/PlantConfig/views/Library/LibraryTreeview/LibraryTreeview.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { Container } from './LibraryTreeview.style';
+
+type LibraryTreeviewProps = {
+    setSelectedLibraryType: (libraryType: string) => void;
+    setSelectedLibraryItem: (libraryItem: string) => void;
+};
+
+const LibraryTreeview = (props: LibraryTreeviewProps): JSX.Element => {
+
+    interface Mode {
```
Comment (kfaln) [inline src/modules/PlantConfig/views/Library/LibraryTreeview/LibraryTreeview.tsx]: Unused

File: src/modules/PlantConfig/style.ts (hunk 1 above)
Comment (arildbjerkenes) [reply src/modules/PlantConfig/style.ts]: Done

File: src/modules/PlantConfig/views/Library/Library.tsx (hunk 2 above)
Comment (arildbjerkenes) [reply src/modules/PlantConfig/views/Library/Library.tsx]: done

File: src/modules/PlantConfig/views/Library/LibraryTreeview/LibraryTreeview.tsx (hunk 3 above)
Comment (arildbjerkenes) [reply src/modules/PlantConfig/views/Library/LibraryTreeview/LibraryTreeview.tsx]: done

**Pull request merged**